While this compiles fine, resulting binary fails to run.
Program received signal EXC_BAD_ACCESS, Could not access memory.
Reason: KERN_PROTECTION_FAILURE at address: 0x00000010
0x003c4130 in pthread_getname_np ()
So just avoid using it on powerpc.

--- src/pulsecore/thread-posix.c	2024-01-13 01:22:09.000000000 +0800
+++ src/pulsecore/thread-posix.c	2024-04-17 17:13:57.000000000 +0800
@@ -182,7 +182,7 @@
 
 #ifdef __linux__
     prctl(PR_SET_NAME, name);
-#elif defined(HAVE_PTHREAD_SETNAME_NP) && defined(OS_IS_DARWIN)
+#elif defined(HAVE_PTHREAD_SETNAME_NP) && defined(OS_IS_DARWIN) && !defined(__POWERPC__)
     pthread_setname_np(name);
 #endif
 }
@@ -201,7 +201,7 @@
             t->name = NULL;
         }
     }
-#elif defined(HAVE_PTHREAD_GETNAME_NP) && defined(OS_IS_DARWIN)
+#elif defined(HAVE_PTHREAD_GETNAME_NP) && defined(OS_IS_DARWIN) && !defined(__POWERPC__)
     if (!t->name) {
         t->name = pa_xmalloc0(17);
         pthread_getname_np(t->id, t->name, 16);
